src/script/auth/page/SetEntropy.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CheckRoundIcon, ContainerXS, H1, Muted, Text} from '@wireapp/react-ui-kit';
+import React, {MouseEvent, PointerEvent, useCallback, useEffect, useState} from 'react';
+
+import {usePausableInterval} from '../../hooks/usePausableInterval';
+import {usePausableTimeout} from '../../hooks/usePausableTimeout';
+import {setEntropyStrings} from '../../strings';
+import {useIntl} from 'react-intl';
+import Canvas from '../component/Canvas';
+
+import {ProgressBar} from '../component/ProgressBar';
+
+interface Props extends React.HTMLProps<HTMLDivElement> {
+  entropy: [number, number][];
+  error: any;
+  onSetEntropy: () => void;
+  setEntropy: React.Dispatch<React.SetStateAction<[number, number][]>>;
+  setError: React.Dispatch<any>;
+}
+
+const SetEntropy = ({setEntropy, entropy, error, setError, onSetEntropy}: Props) => {
```

Comment:
maybe we can come up with a better naming here. `SetEntropy` sound like an action. How about `EntropySetter` or `EntropyGenerator`?